### PR TITLE
Addresses issues in GeoJSON gem

### LINKED ISSUE #32: In building_properties schema for Geojson, null should be included as a building type enumeration. Useful for Mixed Use building type in case not all 4 building types want to be specified.
An additional check can be added when creating building in case building type is added as null and is not mixed use.

### PR DESCRIPTION
**Pull request by tanushree04**
### Pull Request Description

#32 : Supports null building type in the building_properties.json schema.
#37 : Supports `None` and `ShadingOnly` other building types in process_other_buildings method in Helper class. `All` building types no longer supported. 
#38 :  Includes methods to validate geojson file and feature properties against respective schemas, validates geojson and feature properties files(building, district system, region) with additional properties. Includes validation methods that checks fewer required building properties in case `detailed_model_filename` present in GeoJSON file. 
#33 : Fixes bug while implementing zoning, the number of stories created is the same as stories present in the GeoJSON file. Methods for zoning divide the floor plate into core and perimeter zones. 

### Checklist (Delete lines that don't apply)

- [x] Unit tests have been added or updated
- [ ] Documentation has been modified appropriately
- [x] All ci tests pass (green)
- [ ] The [changelog](https://github.com/urbanopt/urbanopt-scenario-gem/blob/develop/CHANGELOG.md) has been updated appropriately
- [x] This branch is up-to-date with develop

### CODE REVIEW

File: lib/urbanopt/geojson/geo_file.rb [hunk 1]
```diff
@@ -69,15 +67,58 @@ def self.from_file(path)
...
+        geojson_file[:features].each do |feature|
+          properties = feature[:properties]
+          type = properties[:type]
+
+          errors = []
+
+          case type
+          when 'Building'
+            # Incase detailed_model_filename present check for fewer properties
+            if feature[:properties][:detailed_model_filename]
```
Comment (tanushree04) [inline lib/urbanopt/geojson/geo_file.rb]: @kflemin This checks whether the detailed_model_filename is present and only checks the id, name properties if it is. Othervise validates the geojson file against the buiding schema.

Comment (kflemin) [reply lib/urbanopt/geojson/geo_file.rb]: looks good!

**Comment by tanushree04** [comment]
@kflemin this is good to be reviewed.

File: lib/urbanopt/geojson/geo_file.rb [hunk 2]
```diff
@@ -119,36 +161,100 @@ def get_feature_by_id(feature_id)
...
+        if strict
+          result['additionalProperties'] = true
+        else
+          result['additionalProperties'] = false
+        end
+        return result      
       end
+
+      strict = true
+      @@geojson_schema = get_geojson_schema(strict)
```
Comment (kflemin) [inline lib/urbanopt/geojson/geo_file.rb]: there are also thermal connectors and thermal junctions...should you add/validate those here too?

File: lib/urbanopt/geojson/schema/district_system_properties.json [hunk 3]
```diff
@@ -79,13 +79,14 @@
...
+        "Central Hot and Chilled Water",
+		          "Central Ambient Water",
+		          "Central Hot and Cold Water", 
+		          "Community Photovoltaic", 
         "Transformer",
         "Transformer with Storage",
-		"Capacitor",
-		"Electrical Substation" 
+		          "Capacitor",
+		          "Electrical Substation"
```
Comment (kflemin) [inline lib/urbanopt/geojson/schema/district_system_properties.json]: remove whitespace here?

Comment (kflemin) [reply lib/urbanopt/geojson/schema/district_system_properties.json]: there are other files with added whitespace.  I think running rubocop will point them out.  thanks!

File: lib/urbanopt/geojson/schema/electrical_connector_properties.json [hunk 4]
```diff
@@ -217,4 +214,4 @@
       ]
     }
   }
-}
+}
```
Comment (kflemin) [inline lib/urbanopt/geojson/schema/electrical_connector_properties.json]: need a newline here

**Pull request merged**